lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyExtension.java

```diff
@@ -77,6 +80,19 @@ public void start(InitializeParams initializeParams, XMLExtensionsRegistry xmlEx
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
+
+        // for each workspace, a file alteration observer is added
+        for (LibertyWorkspace workspace : LibertyProjectsManager.getInstance().getLibertyWorkspaceFolders()) {
+            // checking for any changes in target folder
+            Path libertyPath = new File(workspace.getWorkspaceURI().getPath(),
+                    "target").toPath();
```

Comment:
We have to support Gradle as well. For Gradle projects, we would use `build` folder. I am worried about monitoring the entire target/build folder. I feel like we should be able to be more selective. We only care about the Liberty server folder. In the case that the `liberty-plugin-config.xml` already exists, we can figure out which folder we actually need to monitor (the `serverDirectory`).

Reply:
We were monitoring the target folder because we cannot find serverDirectory without creating liberty-plugin-config.xml file

Watch file is added entirely to support file creation/updating/deletion for config file. But if dev mode is not started, we wont know the serverDirectory so we don't know which path to look for .
Only thing constant in the directory path would be "build" or "target" folder and rest will differ based on serverName.

I will add build folder as well into the watch list